File: src/jk_5/nailed/command/CommandGroup.java
package jk_5.nailed.command;

import jk_5.nailed.Nailed;
import jk_5.nailed.groups.Group;
import jk_5.nailed.players.Player;
import jk_5.nailed.util.EnumColor;
import net.minecraft.server.MinecraftServer;
import net.minecraft.src.*;

import java.util.List;

/**
 * TODO: Edit description
 *
 * @author jk-5
 */
public class CommandGroup extends CommandBase {

    @Override
    public String getCommandName() {
        return "group";
    }

    @Override
    public int getRequiredPermissionLevel() {
        return 3;
    }

    @Override
    public String getCommandUsage(ICommandSender par1ICommandSender) {
        return "/group - Sets the group of a player";
    }

    @Override
    public void processCommand(ICommandSender sender, String[] args) {
        if(args.length == 0){
            throw new WrongUsageException("/group <set>");
        }else if(args.length == 1){
            if(args[0].equals("set")){
                throw new WrongUsageException("/group set <player> <admin:player>");
            }
        }else if(args.length == 2){
            if(args[0].equals("set")){
                Player p = Nailed.playerRegistry.getPlayer(args[1]);
                if(p == null) throw new CommandException("Player " + args[1] + " was not found");
            }
        }else if(args.length == 3){
            if(args[0].equals("set")){
                Player p = Nailed.playerRegistry.getPlayer(args[1]);
                Group g = Nailed.groupRegistry.getGroup(args[2]);
                if(p == null) throw new CommandException("Player " + args[1] + " was not found");
                if(g == null) throw new CommandException("Group " + args[2] + " was not found");
                p.setGroup(g);
                p.sendChatMessage(EnumColor.GREEN + "You were moved to the group " + g.getName());
                sender.sendChatToPlayer(ChatMessageComponent.func_111066_d(EnumColor.GREEN + "Moved player " + p.getUsername() + " to group " + g.getName()));
            }
        }
    }

    @Override
    public List addTabCompletionOptions(ICommandSender sender, String[] args) {
        if(args.length == 1){
            return getListOfStringsMatchingLastWord(args, new String[]{"set"});
        }else if(args.length == 2){
            if(args[0].equals("set")){
                return getListOfStringsMatchingLastWord(args, MinecraftServer.getServer().getAllUsernames());
            }
        }else if(args.length == 3){
            if(args[0].equals("set")){
                return getListOfStringsFromIterableMatchingLastWord(args, Nailed.groupRegistry.getGroups());
            }
        }
        return null;
    }
}

File: src/main/scala/jk_5/nailed/command/CommandGroup.scala
package jk_5.nailed.command

import jk_5.nailed.players.PlayerRegistry
import jk_5.nailed.groups.GroupRegistry
import jk_5.nailed.util.EnumColor

/**
 * No description given
 *
 * @author jk-5
 */
object CommandGroup extends TCommand {
  val commandName = "group"
  this.permissionLevel = 2

  override def getCommandUsage = "/group - Sets the group of a player"
  def processCommand(sender: CommandSender, args: Array[String]) = args.length match{
    case 0 => throw new WrongUsageException("/group <set>")
    case 1 => if(args(0) == "set") throw new WrongUsageException("/group set <player> <admin:player>")
    case 2 => if(args(0) == "set"){
      val p = PlayerRegistry.getPlayer(args(1))
      if(p.isEmpty) throw new CommandException("Player " + args(1) + " was not found")
      else throw new WrongUsageException("/group set <player> <admin:player>")
    }
    case 3 => if(args(0) == "set"){
      val p = PlayerRegistry.getPlayer(args(1))
      val g = GroupRegistry.getGroup(args(2))
      if(p.isEmpty) throw new CommandException("Player " + args(1) + " was not found")
      if(g.isEmpty) throw new CommandException("Group " + args(2) + " was not found")
      p.get.setGroup(g.get)
      p.get.sendChatMessage(EnumColor.GREEN + "You were moved to the group " + g.get.getName)
      sender.sendChatMessage(EnumColor.GREEN + "Moved player " + p.get.getUsername + " to group " + g.get.getName)
    }
    case _ => throw new WrongUsageException("/group <set>")
  }
  override def addAutocomplete(sender: CommandSender, args: Array[String]): AutocompleteList = args.length match{
    case 1 => Command.getMatched(args, "set")
    case 2 => if(args(0) == "set") Command.getMatched(args, Command.getAllUsernames) else null
    case 3 => if(args(0) == "set") Command.getMatched(args, GroupRegistry.getGroups) else null
    case _ => null
  }
}
